### PR TITLE
core: Add `TypeError` handler into `get_graph` of `Runnable`

### PR DESCRIPTION
**Pull request by hmasdev**
# Description

## Problem

`Runnable.get_graph` fails when `InputType` or `OutputType` property raises `TypeError`.

- https://github.com/langchain-ai/langchain/tree/003c98e5b440a420d8dbd5f9fdea08888a4a5a33/libs/core/langchain_core/runnables/base.py#L250-L274
- https://github.com/langchain-ai/langchain/tree/003c98e5b440a420d8dbd5f9fdea08888a4a5a33/libs/core/langchain_core/runnables/base.py#L394-L396

This problem prevents getting a graph of `Runnable` objects whose `InputType` or `OutputType` property raises `TypeError` but whose `invoke`  works well, such as `langchain.output_parsers.RegexParser`, which I have already pointed out in #19792 that a `TypeError` would occur.

## Solution

- Add `try-except` syntax to handle `TypeError` to the codes which get `input_node` and `output_node`.

# Issue
- #19801 

# Twitter Handle
- [hmdev3](https://twitter.com/hmdev3)

### CODE REVIEW

**Comment by ccurme** [comment]
cc @nfcampos

**Comment by hmasdev** [comment]
@baskaryan @efriis @eyurtsev @ccurme @vbarda @hwchase17 
Could anyone review this PR?

File: libs/core/tests/unit_tests/runnables/test_graph.py [hunk 1]
```diff
@@ -687,3 +689,47 @@ def conditional_str_parser(input: str) -> Runnable:
     assert graph.draw_ascii() == snapshot(name="ascii")
     assert graph.draw_mermaid() == snapshot(name="mermaid")
     assert graph.draw_mermaid(with_styles=False) == snapshot(name="mermaid-simple")
+
+
+def test_runnable_get_graph_with_invalid_input_type() -> None:
+    """Test that an error is raised when trying to get graph with invalid input type."""
```
Comment (baskaryan) [inline libs/core/tests/unit_tests/runnables/test_graph.py:695]: is this mean to say "is *not* raised" ?

Comment (hmasdev) [reply libs/core/tests/unit_tests/runnables/test_graph.py]: yes

**Pull request merged**